### PR TITLE
feat(secrets): Pass in secrets

### LINKED ISSUE #929: Add support for passing `credentials`  for RDS directly
This way I could create the credentials in secrets manager and pass that in, but we able to refer to that later as well.

### PR DESCRIPTION
**Pull request by praveenperera**
- Fixes #929
- Also allows you to pass in `securityGroups` 
- Replaced deprecated `PRIVATE_WITH_NAT`

### CODE REVIEW

**Comment by neilkuan** [comment]
hi @praveenperera , could you run npx projen locally and yarn build again.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (neilkuan) [inline .vscode/settings.json:1]: I think this not need to commit. :)

**Review by neilkuan — APPROVED** [review]
Thanks for your Contributions.

**Comment by praveenperera** [comment]
@neilkuan thanks for merging it in quick, and thanks for the package

**Pull request merged**